products/Product.js

```diff
@@ -692,6 +693,12 @@ export default class Product {
         filteredProductDefinitions.push(currentProductDefinition)
     }
 
+    if (!includeMauiOnly) {
+      filteredProductDefinitions = filteredProductDefinitions.filter(
+        (prodDef) => !prodDef.isMauiOnly()
```

Comment:
do not modify existing objects. this obscures what the array contains. this should be named `mauiOnlyProductDefinitions`